### PR TITLE
[CDAP-2426] Add adapter name to workflow logging context

### PR DESCRIPTION
**Pull request by bdmogal**
Jira: [CDAP-2426](https://issues.cask.co/browse/CDAP-2426)
Build: http://builds.cask.co/browse/CDAP-RBT269

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java [hunk 1]
```diff
@@ -107,10 +110,15 @@
...
-                                                options.getArguments().getOption(ProgramOptionConstants.RUN_ID));
+    this.lock = new ReentrantLock();
+    this.condition = lock.newCondition();
+    String adapterSpec = arguments.getOption(ProgramOptionConstants.ADAPTER_SPEC);
+    String adapterName = null;
+    if (adapterSpec != null) {
+      adapterName = GSON.fromJson(adapterSpec, AdapterDefinition.class).getName();
+    }
+    this.loggingContext = new WorkflowLoggingContext(program.getNamespaceId(), program.getApplicationId(), program.getName(),
+                                                arguments.getOption(ProgramOptionConstants.RUN_ID), adapterName);
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java]: fix alignment

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/LogHandlerTestRun.java [hunk 2]
```diff
@@ -207,6 +207,49 @@ public void testAdapterLogsPrev() throws Exception {
   }
 
   @Test
+  public void testWorkflowLogs() throws Exception {
+    testLogs("testTemplate1", "workflows", "testWorkflow1", MockLogReader.TEST_NAMESPACE, "testAdapter1");
+    try {
+      // in default namespace, we did not produce logs through adapters
+      testNext("testTemplate1", "workflows", "testWorkflow1", true, Constants.DEFAULT_NAMESPACE, "testAdapter1");
+      Assert.fail();
+    } catch (AssertionError e) {
```
Comment (gokulavasan) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/LogHandlerTestRun.java]: Here we call Assert.fail() to throw an AssertionError and then we catch it and do nothing. So what is the test trying to test? Same comment for this pattern used below.

**Comment by gokulavasan** [comment]
Couple of comments. Rest LGTM :+1:

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java (hunk 1 above)
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java]: Fixed.

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/LogHandlerTestRun.java (hunk 2 above)
Comment (bdmogal) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/LogHandlerTestRun.java]: Great catch! This was a pattern throughout this test. Basically, we're expecting an ``AssertionError`` on line 214 above.  But throwing an ``Assert.fail()`` in the try defeats the purpose. I've fixed it in the next commit by moving this verification to the end of each test, and calling a return from the catch clause, then adding an ``Assert.fail()`` at the end of the test... also fixed it throughout this class.

**Comment by bdmogal** [comment]
@gokulavasan addressed comments in the second commit

**Comment by gokulavasan** [comment]
LGTM :+1:

**Comment by bdmogal** [comment]
Build passed. Squashed commits. Merging.

**Pull request merged**